### DISCUSSION TITLE
Custom Models

### DISCUSSION THREAD

**Discussion by sorgfresser** [aarnphm/whispercpp#81] [General]
Thanks for your great work!
I'm curious: the plain whispercpp supports custom ggml-models. Is there any way to use it with python-bindings? I'm thinking from_pretrained could get a local path to a `ggml-model.bin`.
Is that already possible / feasible?

**Comment by aarnphm** [comment]
I haven't implemented that yet. But it is pretty easy to support. We can do that in the Python side. Right now it handles the saved weights [here](https://github.com/aarnphm/whispercpp/blob/main/src/whispercpp/utils.py).

Feel free to open a PR for it. One thing to note is that we just have to handle `from_pretrained` to also accepts arbitrary path.

**Reply by sorgfresser** [reply]
PR is created. Not so sure about arbitrary paths. I'm checking using `os.path.isfile` right now.

**Reply by aarnphm** [reply]
you can probably do `os.path.abspath`
